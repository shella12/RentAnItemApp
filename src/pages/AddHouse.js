import { useDispatch } from 'react-redux';
import { useState } from 'react';
import FlashMessage from '../componenets/FlashMessage';
import { addHouse } from '../redux/house/house';
import Navbar from '../componenets/navbar/Navbar';

const AddHouse = () => {
  const dispatch = useDispatch();
  const [name, setName] = useState('');
  const [price, setPrice] = useState('');
  const [description, setDescription] = useState('');
  const [picture, setPicture] = useState('');
  const [owner, setOwner] = useState('');
  const [status, setStatus] = useState(false);
  const [message, setMessage] = useState('');

  const handleSubmit = (event) => {
    event.preventDefault();
    const form = event.target;
    const data = new FormData();
    data.append('house[name]', form.name.value);
    data.append('house[price]', form.price.value);
    data.append('house[owner_name]', form.owner.value);
    data.append('house[description]', form.description.value);
    data.append('house[picture]', form.picture.files[0]);
    setStatus(true);
    dispatch(addHouse(data));
    setName('');
    setPrice('');
    setDescription('');
    setPicture('');
    setOwner('');
    setMessage('your house has been added successfully!');
  };

  return (
    <>
      <Navbar title="New House" />
      <section className="add-house-section">
        <div className="backdrop column flex-center">
          <h1>Add a house</h1>
          <p>
            Turn the key and walk through the door.
            This new house is all you ever wanted and more.
            The best journey takes you home
          </p>
          {status && <FlashMessage message={message} duration={5000} />}
          <form className="column" onSubmit={handleSubmit} onChange={() => setStatus(false)}>
            <input
              className="input-text"
              name="name"
              type="text"
              placeholder="House type e.g pent house, cottage etc..."
              value={name}
              onChange={(e) => setName(e.target.value)}
              required
            />
            <input className="input-text" name="price" min="1" type="number" placeholder="Price e.g $30,00,0000..." value={price} onChange={(e) => setPrice(e.target.value)} required />
            <input className="input-text" name="owner" type="text" placeholder="owner name e.g John doe" value={owner} onChange={(e) => setOwner(e.target.value)} required />
            <input className="input-text" name="picture" type="file" placeholder="Picture url" value={picture} onChange={(e) => setPicture(e.target.value)} required />
            <textarea className="input-textarea" name="description" placeholder="write a description of the house ..." value={description} onChange={(e) => setDescription(e.target.value)} required />
            <button className="btn" type="submit">Add a house</button>
          </form>
        </div>
      </section>
    </>
  );
};

export default AddHouse;
