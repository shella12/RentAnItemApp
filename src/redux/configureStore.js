import { configureStore } from '@reduxjs/toolkit';
import favoriteReducer from './favorites/favoriteReducer';
import housesSlice from './house/house';

const store = configureStore({
  reducer: {
    housesSlice,
    favorite: favoriteReducer,
  },
});

export default store;
