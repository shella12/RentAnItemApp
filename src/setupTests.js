import '@testing-library/jest-dom';
import fetchMock from 'fetch-mock';

export const testListHouses = [{
  id: 1, name: 'Pent house', price: '300000.0', picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/5bb4ea2c-4ae5-45b8-9083-aeafbfbcf287', description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.", owner_name: 'susan perry', created_at: '2023-05-12T12:38:59.989Z', updated_at: '2023-05-12T12:38:59.989Z',
}, {
  id: 6, name: 'Mansion', price: '870000.0', picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/ac143109-1ae3-4885-a650-808b3bdb6cda', description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.", owner_name: 'Kim buffey', created_at: '2023-05-12T12:39:00.010Z', updated_at: '2023-05-12T12:39:00.010Z',
}];

export const testListFavoriteHouses = [
  {
    id: 1,
    name: 'Pent house',
    price: '300000.0',
    picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/5bb4ea2c-4ae5-45b8-9083-aeafbfbcf287',
    description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.",
    owner_name: 'susan perry',
    created_at: '2023-05-12T12:38:59.989Z',
    updated_at: '2023-05-12T12:38:59.989Z',
  },
];

export const fetchFavoriteHouseData = (userID) => {
  fetchMock.getOnce(`https://renthousebackend.onrender.com/api/v1/users/${userID}/favorite_houses`, {
    headers: { 'content-type': 'application/json' },
    body: testListFavoriteHouses,
  });
};

export const fetchHouseData = () => {
  fetchMock.getOnce('https://renthousebackend.onrender.com/api/v1/houses', {
    headers: { 'content-type': 'application/json' },
    body: testListHouses,
  });
};
