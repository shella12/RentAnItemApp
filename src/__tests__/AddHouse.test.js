import renderer, { act } from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import { waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { fetchHouseData } from '../setupTests';
import configureStore from '../redux/configureStore';
import AddHouse from '../pages/AddHouse';

const store = configureStore;
fetchHouseData();

describe('test render', () => {
  test('Add House page should match snapshoot', async () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <React.StrictMode>
          <Provider store={store}>
            <MemoryRouter initialEntries={['/houses/add']}>
              <AddHouse />
            </MemoryRouter>
          </Provider>
        </React.StrictMode>,
      );
    });

    await waitFor(() => {
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
