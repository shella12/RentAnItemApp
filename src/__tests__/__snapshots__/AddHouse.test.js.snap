// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render Add House page should match snapshoot 1`] = `
Array [
  <div
    className="navbar"
  >
    <div
      className="hamburger"
    >
      <div
        aria-expanded={false}
        className="hamburger-react"
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
            "height": "48px",
            "outline": "none",
            "position": "relative",
            "transform": "none",
            "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
            "userSelect": "none",
            "width": "48px",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            Object {
              "background": "#7e7d7f",
              "borderRadius": "9em",
              "height": "2px",
              "left": "14px",
              "position": "absolute",
              "top": "17px",
              "transform": "none",
              "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
              "width": "20px",
            }
          }
        />
        <div
          style={
            Object {
              "background": "#7e7d7f",
              "borderRadius": "9em",
              "height": "2px",
              "left": "14px",
              "position": "absolute",
              "top": "23px",
              "transform": "none",
              "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
              "width": "20px",
            }
          }
        />
        <div
          style={
            Object {
              "background": "#7e7d7f",
              "borderRadius": "9em",
              "height": "2px",
              "left": "14px",
              "position": "absolute",
              "top": "29px",
              "transform": "none",
              "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
              "width": "20px",
            }
          }
        />
      </div>
      <nav
        className="navlinks"
      >
        <div
          className=" column mob-profile"
        >
          <img
            alt="profile"
            className="profile-photo mob-profile-photo"
            src="profile-photo.jpeg"
          />
          <p
            className="username mob-username"
          >
            Username
          </p>
          <p />
        </div>
        <a
          activeclassname="active"
          aria-current="page"
          aria-label="Houses link"
          className="active"
          href="/houses"
          onClick={[Function]}
        >
          Houses
        </a>
        <a
          activeclassname="active"
          aria-label="My Favorites link"
          className=""
          href="/houses/favorites"
          onClick={[Function]}
        >
          My Favorites
        </a>
        <a
          activeclassname="active"
          aria-current="page"
          aria-label="Add house link"
          className="active"
          href="/houses/add"
          onClick={[Function]}
        >
          Add house
        </a>
        <a
          activeclassname="active"
          aria-label="Delete house link"
          className=""
          href="/houses/delete"
          onClick={[Function]}
        >
          Delete house
        </a>
        <hr
          className="divider"
        />
        <a
          activeclassname="active"
          aria-current="page"
          aria-label="Log Out link"
          className="active"
          href="/houses/add"
          onClick={[Function]}
        >
          Log Out
        </a>
      </nav>
    </div>
    <h1>
      New House
    </h1>
    <svg
      className="search-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
      />
    </svg>
  </div>,
  <section
    className="add-house-section"
  >
    <div
      className="backdrop column flex-center"
    >
      <h1>
        Add a house
      </h1>
      <p>
        Turn the key and walk through the door. This new house is all you ever wanted and more. The best journey takes you home
      </p>
      <form
        className="column"
        onChange={[Function]}
        onSubmit={[Function]}
      >
        <input
          className="input-text"
          name="name"
          onChange={[Function]}
          placeholder="House type e.g pent house, cottage etc..."
          required={true}
          type="text"
          value=""
        />
        <input
          className="input-text"
          min="1"
          name="price"
          onChange={[Function]}
          placeholder="Price e.g $30,00,0000..."
          required={true}
          type="number"
          value=""
        />
        <input
          className="input-text"
          name="owner"
          onChange={[Function]}
          placeholder="owner name e.g John doe"
          required={true}
          type="text"
          value=""
        />
        <input
          className="input-text"
          name="picture"
          onChange={[Function]}
          placeholder="Picture url"
          required={true}
          type="file"
          value=""
        />
        <textarea
          className="input-textarea"
          name="description"
          onChange={[Function]}
          placeholder="write a description of the house ..."
          required={true}
          value=""
        />
        <button
          className="btn"
          type="submit"
        >
          Add a house
        </button>
      </form>
    </div>
  </section>,
]
`;
